concordia/static/js/contribute.js

```diff
@@ -415,3 +415,23 @@ $tagEditor
 
         displayMessage('error', message, 'tags-save-result');
     });
+
+var $copyUrlButton = $('#copy-url-button');
+$copyUrlButton.tooltip();
```

Comment:
Currently the tooltip displays on click no matter what actually happens. I'm wondering whether that should be changed to manual activation and then the JavaScript API would be used to display it after the clipboard operation is successful:

```suggestion
$copyUrlButton.tooltip({trigger: "manual"});
```

Part of why I'm thinking about this is that while most browsers support the clipboard API past abuses have lead to a number of restrictions:

https://caniuse.com/#feat=clipboard

That makes me wonder whether the logic basically needs to be something like putting the current body of the click even handler into a try block which would display either a success message (with auto-hiding) or an error message (which doesn't go away until dismissed) after the `execCommand` call.

Reply:
This is done, except for making the error message stay until dismissal... working on that part.